### ISSUE TITLE
ENH: Add built-in hyperparameter tuning

### ISSUE THREAD

**Issue by benkambs** [e-dyn/gloria#55]
**Is your feature request related to a problem? Please describe.**
Gloria currently mirrors Prophet’s ``cross_validation()`` and ``performance_metrics()`` helpers, but selecting optimal hyperparameter such as prior_scales or used seasonalities are still left to the user. This is error-prone, slow to prototype, and makes experiments harder to reproduce and share.

**Describe the solution you'd like**
An additional helper function named something like ``parameter_scan()`` or ``evaluate_parameters()`` residing in the ``diagnostics`` module could help streamlining Gloria workflows, for instance like the following:

``` python
from gloria import Gloria, evaluate_parameters

# Set up a reference model
m = Gloria(...)
m.add_seasonality(...)
m.add_external_regressor(...)

# Load data to perform the optimization on
data = pd.read_csv(...)

# Define search grid for parameters
param_grid = {
    "changepoint_prior_scale": [0.01, 0.1, 0.5],
    "seasonality_prior_scale": [1.0, 5.0],
    "holidays_prior_scale": [5.0, 10.0],
}

# Iterate through all parameter combinations
results = evaluate_parameters(
    model=m,
    data=data,
    param_grid=param_grid,
    method="grid search",
    horizon="90 days",
    period="30 days",
    initial="365 days",
    metric="mape"
)

# Pick the best performing set
best_params = results.loc[results["rank"] == 1, results.columns.str.startswith("param_")]

```

- The reference model ``m`` defines all fixed parameters.
- ``param_grid`` defines a search grid similar to the one used in scikit learn's [GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html)
- (optional) a ``method`` argument let`s the user decide how to perform the search (similar to scikit learn's grid search, halving grid search, or randomized search)
- the output of ``evaluate_parameters()`` is a dataframe containing the tested parameter combinations, how they performed according to the given metric, as well as a ``rank`` column to quickly find the best parameter set

**Additional context**

- As hyper parameter tuning is known to be slow, options for parallelizing the task should be considered
